cmd/orca-tasks/task_watcher.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2019, Google, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package orca_tasks
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/spinnaker/spin/cmd/gateclient"
+)
+
+// WaitForSuccessfulTask observes an Orca task to see if it completed successfully.
+func WaitForSuccessfulTask(gateClient *gateclient.GatewayClient, taskRef map[string]interface{}, maxAttempts int) error {
+	id := idFromTaskRef(taskRef)
+	task, resp, err := gateClient.TaskControllerApi.GetTaskUsingGET1(gateClient.Context, id)
+
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		return fmt.Errorf("Encountered an error saving application, status code: %d\n", resp.StatusCode)
+	}
+	if !taskSucceeded(task) {
```

Comment:
It looks like if the task is still running, we won't even get to the retry logic...should this be removed entirely and instead fallback to the `taskSucceeded` check after we know the task is complete?

Reply:
Agreed, I just removed all checks here and delegate to the normal while `for` loop.